src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs

```diff
@@ -32,33 +36,109 @@ public class TestJsonCommand : PSCmdlet
         /// then validates the JSON against the schema. Before testing the JSON string,
         /// the cmdlet parses the schema doing implicitly check the schema too.
         /// </summary>
-        [Parameter(Position = 1)]
+        [Parameter(Position = 1, ParameterSetName = TestJsonCommand.SchemaStringParameterSet)]
         [ValidateNotNullOrEmpty()]
         public string Schema { get; set; }
 
+        /// <summary>
+        /// A path to the file containg schema to validate the JSON against.
+        /// This is optional parameter.
+        /// If the parameter is absent the cmdlet only attempts to parse the JSON string.
+        /// If the parameter present the cmdlet attempts to parse the JSON string and
+        /// then validates the JSON against the schema. Before testing the JSON string,
+        /// the cmdlet parses the schema doing implicitly check the schema too.
+        /// </summary>
+        [Parameter(Position = 1, ParameterSetName = TestJsonCommand.SchemaPathParameterSet)]
+        [ValidateNotNullOrEmpty()]
+        public string SchemaPath { get; set; }
+
         private JsonSchema _jschema;
 
         /// <summary>
-        /// Prepare an JSON schema.
+        /// Prepare a JSON schema.
         /// </summary>
         protected override void BeginProcessing()
         {
-            if (Schema != null)
+            string resolvedpath = string.Empty;
+
+            try
             {
-                try
+                if (Schema != null)
                 {
-                    _jschema = JsonSchema.FromJsonAsync(Schema).Result;
+                    try
+                    {
+                        _jschema = JsonSchema.FromJsonAsync(Schema).Result;
+                    }
+                    // Even if only one exception is thrown, it is still wrapped in an AggregateException exception
+                    // https://docs.microsoft.com/en-us/dotnet/standard/parallel-programming/exception-handling-task-parallel-library
+                    catch (AggregateException ae)
+                    {
+                        // Process all exceptions in the AggregateException
+                        ae.Handle(i =>
```

Comment:
i - please use more informative name.

Reply:
Renamed to `ie` as in `inner exception`. Would this be ok?

Reply:
`Handle` is now invoking private method instead of delegates, so variables are removed..